src/lib.rs

```diff
@@ -0,0 +1,196 @@
+extern crate failure;
+extern crate object;
+extern crate uuid;
+
+#[cfg(target_os="macos")]
+extern crate CoreFoundation_sys as cf;
+
+use failure::Error;
+use object::{DebugFileInfo, File, Object};
+use std::path::{Path, PathBuf};
+use uuid::Uuid;
+
```

Comment:
It might be worth mentioning that this code is basically an implementation of DBGCopyFullDSYMURLForUUID from the DebugSymbols framework

Reply:
Are there any docs on `DebugSymbols`? When I originally looked into it details were scarce.

Reply:
There are not. I wrote https://github.com/jrmuizel/dsym-find/blob/master/dsym-find.c based on lldb.

Reply:
Thanks! I suspect that `DebugSymbols` does more than what this crate currently does under the hood, [given the docs](http://lldb.llvm.org/symbols.html). Supporting those things (like file-mapped UUID directories) seems like it'd be useful for quicker lookups and better interop with lldb.

Reply:
So should this just use `DebugSymbols` then?

Reply:
I'm a bit wary of relying on a private framework even if it does more of what we want. I think supporting *most* of what `DebugSymbols` does would be fairly straightforward, and I'd love to have some of that work cross-platform so it's feasible to write tools that work on Mach-O binaries and are usable on other platforms.